service/raft/group0_state_machine.cc

```diff
@@ -123,16 +123,49 @@ bool should_flush_system_topology_after_applying(const mutation& mut, const data
     return false;
 }
 
-future<> write_mutations_to_database(storage_proxy& proxy, gms::inet_address from, utils::chunked_vector<canonical_mutation> cms) {
+struct client_routes_update {
+    std::vector<sstring> connection_ids;
+    std::vector<sstring> host_ids;
+};
+
+void collect_client_routes_update(schema_ptr s, mutation& mut, storage_proxy& proxy, client_routes_update& client_routes_update) {
```

Comment:
Why not return `client_routes_update` as a result of `collect_client_routes_update`? More natural than out parameter.